### PR TITLE
Add a dialog to configure pipeline's runtime resources

### PR DESCRIPTION
**Pull request by Karakatiza666**
Fix https://github.com/feldera/feldera/issues/939

Blocked by https://github.com/feldera/feldera/issues/1026

### CODE REVIEW

**Comment by Karakatiza666** [comment]
image
image
image

**Comment by lalithsuresh** [comment]
@Karakatiza666 the resources dialog is only relevant in the cloud form factor, so I'm not sure whether we should have the dialog at all in other form factors.

**Comment by Karakatiza666** [comment]
@lalithsuresh How can WebConsole differentiate between current deployment type?

**Comment by lalithsuresh** [comment]
Is there a reason we're doing this with a separate gear button than where we edit the pipeline?

**Comment by Karakatiza666** [comment]
I thought It would be convenient to be able to update config straight from pipeline list page, let me know If it's better to not add button there

**Comment by gz** [comment]
>@Karakatiza666 the resources dialog is only relevant in the cloud form factor, so I'm not sure whether we should have the dialog at all in other form factors.

I think we can set e.g., workers for the pipeline too in non-cloud deployment so having some (reduced) dialog in local deployment is useful

**Comment by gz** [comment]
I'd swap the settings and delete button so the delete button is at the end of the icon list

**Comment by gz** [comment]
I'd use min/max sliders for workers/cpu/memory

**Comment by gz** [comment]
And we should name units as standard SI units.. e.g., MB if we do `**10` (not Mb) or MiB if we use `**`

**Comment by Karakatiza666** [comment]
image

**Comment by lalithsuresh** [comment]
I'm not sure a log slider is the right UX for this. There will be a cluster towards some standard sizes over time, and a log scale makes it hard to be precise at the higher sizes. I'd just keep the simple form you had before.

**Comment by Karakatiza666** [comment]
The values can still be input via number fields for precision, and the log scale is configurable

**Comment by gz** [comment]
The slider solves a bunch of invalid input problems (min>max, not a number, too big values etc.) by avoiding them entirely IMO overall better UX

but yea let's not do log-scale

**Comment by snkas** [comment]
Some minor remarks:
- For the future: we might also want to add a hover-over hint with information about how the minimum and maximum are applied
- Instead of "null" as the gray value when nothing is entered, possibly "None"? Or a dash? Or something else?
- What are the default values shown for memory and CPU min? Is maximum optional?

**Comment by Karakatiza666** [comment]
As far as I am aware there are no server-side default values other than `workers` which default to 8, and min/max that can be entered In the UI are displayed on slider's ends.
Please let me know if limits for values are reasonable, and any comments that I could embed as a tooltip message, e.g. "Logical CPUs to be shared between all pipeline workers".

**Comment by Karakatiza666** [comment]
I will update the form once backend provides min and max for each field

**Comment by Karakatiza666** [comment]
Right now maximum values are enforced; I can change it so that maximum values are only applied to the slider limit, and a greater value can be input through the text field

**Comment by mihaibudiu** [comment]
Is this PR still relevant?

**Comment by lalithsuresh** [comment]
It is, but ended up lower on the priority.

**Comment by Karakatiza666** [comment]
Workers: 1 - 64
CPU: 1 - 64
Memory 100 MB - 64 GB
Storage 1 GB - 1 TB

undefined:
workers not possible
cpus -> server default
mem -> server default
storage -> server default

This configuration will apply

**Comment by Karakatiza666** [comment]
image
image
image

**Comment by Karakatiza666** [comment]
image
image

File: CHANGELOG.md [hunk 1]
```diff
@@ -400,6 +400,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   ([#996](https://github.com/feldera/feldera/pull/996))
 - pipeline-manager now exposes a scrape endpoint for metrics, starting with the compiler service
   ([#1031](https://github.com/feldera/feldera/pull/1031))
+- WebConsole: Add a dialog to configure pipeline's runtime resources
+  ([#939](https://github.com/feldera/feldera/issues/939))
```
Comment (snkas) [inline CHANGELOG.md:404]: It's PR 1027?

**Review by snkas — APPROVED** [review]

**Comment by gz** [comment]
This looks good to me. 

The thing I am not sure about is the "storage class", do we expect the user to know what to write there (I didn't fwiw). Maybe we keep it simple and just avoid this parameter and the "operator" that runs the thing has to decide when setting up the pipeline manager? But then I don't know if there is some other reasoning for this parameter I'm missing out on.

**Comment by lalithsuresh** [comment]
@gz this is something @snkas and I discussed. This is aimed at operators. It's only relevant for the Enterprise Edition to pick the kind of storage we use on Kubernetes.

**Comment by gz** [comment]
Ok so maybe it should be a select box with the available options (and there is only one option for the local form factor which is default?)

**Comment by lalithsuresh** [comment]
@gz yes that is indeed the end goal, but this was easier to get started with. Currently, we don't have an API to signal the form factor from the pipeline manager, which is why we went with this rudimentary option first.

**Comment by snkas** [comment]
Yes, for now just a string, but once we settled on what storage performance classes we want to define, I'll turn it into an enumeration in a later PR.

**Pull request merged**